src/io/import_off.cc

```diff
@@ -2,24 +2,234 @@
+    return std::make_unique<PolySet>(3);
+  }
+
+  while (line.length() == 0) {
+    if (!getline_clean("bad header: end of file"))
+      return std::make_unique<PolySet>(3);
+  }
+  std::vector<std::string> words;
+
+  if (has_ndim) {
+    boost::split(words, line, boost::is_any_of(" \t"), boost::token_compress_on);
+    if (f.eof() || words.size() < 1) {
+      AsciiError("bad header: missing Ndim");
+      return std::make_unique<PolySet>(3);
+    }
+    line = line.erase(0, words[0].length() + ((words.size() > 1) ? 1 : 0));
+    try {
+      dimension = boost::lexical_cast<unsigned int>(words[0]) + dimension - 3;
+    } catch (const boost::bad_lexical_cast& blc) {
+      AsciiError("bad header: bad data for Ndim");
+      return std::make_unique<PolySet>(3);
+    }
+  }
+
+  PRINTDB("Header flags: N:%d C:%d ST:%d Ndim:%d B:%d", has_normals % has_color % has_textures % dimension % is_binary);
+
+  if (dimension != 3) {
+    AsciiError("unhandled vertex dimensions");
+    return std::make_unique<PolySet>(3);
+  }
+
+  while (line.length() == 0) {
+    if (!getline_clean("bad header: end of file"))
+      return std::make_unique<PolySet>(3);
+  }
+
+  boost::split(words, line, boost::is_any_of(" \t"), boost::token_compress_on);
+  if (f.eof() || words.size() < 3) {
+    AsciiError("bad header: missing data");
+    return std::make_unique<PolySet>(3);
+  }
+
+  unsigned long vertices_count;
+  unsigned long faces_count;
+  unsigned long edges_count;
+  unsigned long vertex = 0;
+  unsigned long face = 0;
+  try {
+    vertices_count = boost::lexical_cast<unsigned long>(words[0]);
+    faces_count = boost::lexical_cast<unsigned long>(words[1]);
+    edges_count = boost::lexical_cast<unsigned long>(words[2]);
+    (void)edges_count; // ignored
+  } catch (const boost::bad_lexical_cast& blc) {
+    AsciiError("bad header: bad data");
+    return std::make_unique<PolySet>(3);
+  }
+
+  if (f.eof() || vertices_count < 1 || faces_count < 1) {
+    AsciiError("bad header: not enough data");
+    return std::make_unique<PolySet>(3);
+  }
+
+  PRINTDB("%d vertices, %d faces, %d edges.", vertices_count % faces_count % edges_count);
+
+  auto ps = std::make_unique<PolySet>(3);
+  ps->vertices.reserve(vertices_count);
+  ps->indices.reserve(faces_count);
+
+  while ((!f.eof()) && (vertex++ < vertices_count)) {
+    if (!getline_clean("reading vertices: end of file"))
+      return std::make_unique<PolySet>(3);
+    if (line.length() == 0)
+      continue;
+
+    boost::trim(line);
+    boost::split(words, line, boost::is_any_of(" \t"), boost::token_compress_on);
+    if (words.size() < 3) {
+      AsciiError("can't parse vertex: not enough data");
+      return std::make_unique<PolySet>(3);
+    }
+
+    try {
+      Vector3d v;
+      for (int i = 0; i < 3; i++) {
+        v[i]= boost::lexical_cast<double>(words[i]);
+      }
+      //PRINTDB("Vertex[%ld] = { %f, %f, %f }", vertex % v[0] % v[1] % v[2]);
+      int o = dimension;
```

Comment:
Avoid single-character variable names unless it's very obvious what it means.
I can't really tell what the below code does

Reply:
Oh, that's the offset into the split words, but I changed the code to just reuse `i` since we increment it anyway.